core/graphelier-service/models/orderbooks.go

```diff
@@ -51,37 +53,36 @@ func (orderbook *Orderbook) applyNewOrder(message *Message) {
 		return
 	}
 	order := &Order{message.OrderID, message.ShareQuantity}
-	if message.Direction == -1 {
-		index, found := getLevelIndex(orderbook.Asks, message)
-		if !found {
+	index, found := orderbook.getLevelIndex(message)
+	if !found {
+		if message.Direction == Asks {
 			level := &Level{message.Price, []*Order{order}}
 			orderbook.Asks = append(orderbook.Asks, nil)
 			copy(orderbook.Asks[index+1:], orderbook.Asks[index:])
 			orderbook.Asks[index] = level
-		} else {
-			orderbook.Asks[index].Orders = append(orderbook.Asks[index].Orders, order)
-		}
-	} else if message.Direction == 1 {
-		index, found := getLevelIndex(orderbook.Bids, message)
-		if !found {
+		} else if message.Direction == Bids {
 			level := &Level{message.Price, []*Order{order}}
 			orderbook.Bids = append(orderbook.Bids, nil)
 			copy(orderbook.Bids[index+1:], orderbook.Bids[index:])
 			orderbook.Bids[index] = level
-		} else {
+		}
+	} else {
+		if message.Direction == Asks {
+			orderbook.Asks[index].Orders = append(orderbook.Asks[index].Orders, order)
+		} else if message.Direction == Bids {
 			orderbook.Bids[index].Orders = append(orderbook.Bids[index].Orders, order)
 		}
 	}
 }
 
 func (orderbook *Orderbook) applyModify(message *Message) {
-	orders, orderIndex := getOrders(orderbook, message)
+	orders, orderIndex := orderbook.getOrders(message)
 	if orderIndex < 0 || orders == nil {
 		return
 	}
 
 	o := (*orders)[orderIndex]
-	if message.ShareQuantity >= 0 && message.ShareQuantity < o.Quantity {
+	if message.ShareQuantity < o.Quantity {
```

Comment:
I think this is a revert of a [commit on master](https://github.com/Lercerss/graphelier/commit/45bd2ba56923ee5bcfed03e2ccd43d73b769eb30#diff-bc20862332dac6afb0d57539c82e46e1R84), I am not sure if we want that

Reply:
I think this is a check for a uint being less than 0 which isn’t possible

Reply:
The importer assumes that the share quantity can be negative on modify messages. This was added so that the assumptions remain consistent. If the share quantity on messages in a uint, can you make it an int?

Reply:
Oh ya sure I’ll change that